src/routes/teams.rs

```diff
@@ -1,25 +1,227 @@
+                name: new_member.name.clone(),
+                role: new_member.role.clone(),
+                permissions: new_member.permissions,
+                accepted: false,
+            }
+            .insert(&mut transaction)
+            .await
+            .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+            transaction
+                .commit()
+                .await
+                .map_err(|e| ApiError::DatabaseError(e.into()))?;
+
+            Ok(HttpResponse::Ok().body(""))
+        } else {
+            Err(ApiError::AuthenticationError)
+        }
+    } else {
+        Err(ApiError::AuthenticationError)
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct EditTeamMember {
+    pub permissions: Option<Permissions>,
+    pub role: Option<String>,
+}
+
+#[patch("{id}/members/{user_id}")]
+pub async fn edit_team_member(
+    req: HttpRequest,
+    info: web::Path<(TeamId, UserId)>,
+    pool: web::Data<PgPool>,
+    edit_member: web::Json<EditTeamMember>,
+) -> Result<HttpResponse, ApiError> {
+    let ids = info.into_inner();
+    let id = ids.0.into();
+    let user_id = ids.1.into();
+
+    let current_user = get_user_from_headers(req.headers(), &**pool)
+        .await
+        .map_err(|_| ApiError::AuthenticationError)?;
+    let team_member = TeamMember::get_from_user_id(id, current_user.id.into(), &**pool).await?;
+
+    if let Some(member) = team_member {
+        if member.permissions.contains(Permissions::EDIT_MEMBER)
+            && edit_member.role.as_deref() != Some(crate::models::teams::OWNER_ROLE)
+        {
+            if let Some(new_permissions) = edit_member.permissions {
+                if !member.permissions.contains(new_permissions) {
+                    return Err(ApiError::AuthenticationError);
+                }
+            }
+
+            TeamMember::edit_team_member(
+                id,
+                user_id,
+                edit_member.permissions.map(|x| x.bits() as i64),
+                edit_member.role.clone(),
+                None,
+                &**pool,
+            )
+            .await?;
+
+            Ok(HttpResponse::Ok().body(""))
+        } else {
+            Err(ApiError::AuthenticationError)
+        }
+    } else {
+        Err(ApiError::AuthenticationError)
+    }
+}
+
+#[delete("{id}/members/{user_id}")]
+pub async fn remove_team_member(
+    req: HttpRequest,
+    info: web::Path<(TeamId, UserId)>,
+    pool: web::Data<PgPool>,
+) -> Result<HttpResponse, ApiError> {
+    let ids = info.into_inner();
+    let id = ids.0.into();
+    let user_id = ids.1.into();
+
+    let current_user = get_user_from_headers(req.headers(), &**pool)
+        .await
+        .map_err(|_| ApiError::AuthenticationError)?;
+    let team_member = TeamMember::get_from_user_id(id, current_user.id.into(), &**pool).await?;
+
+    if let Some(member) = team_member {
+        let delete_member_option = TeamMember::get_from_user_id(id, user_id, &**pool).await?;
```

Comment:
This looks the exact same as the query for the team member above.  Are we ever actually checking that members have `accepted = true` before they can use the permissions for that team member?  The `permissions` bitset for a non accepted team member should be treated like it has no permissions.

Reply:
`get_from_user_id` can only return accepted team members, so this route cannot remove pending team member requests.  We need to add a new function for getting potentially pending team members and use that here;  `get_from_user_id` must stay the same and only return accepted members because it is used for checking permissions.